toman/README.md

```diff
@@ -0,0 +1,122 @@
+# Project Setup
+
+## Step 1 – Installing the Components from the Ubuntu Repositories
+
+First you will install the essential components. This includes pip, the Python package manager for installing and managing Python components, and also the database software with its associated libraries.
+
+You will be using Python 3, which ships with Ubuntu 20.04. Start the installation by typing:
+
+    ➜ ~ sudo apt update
+    ➜ ~ sudo apt install python3-venv python3-pip python3-dev libpq-dev postgresql postgresql-contrib
+
+With the installation out of the way, you can move on to the database.
+
+## Step 2 – Creating a Database and Database User
+
+By default, Postgres uses an authentication scheme called “peer authentication” for local connections. Basically, this means that if the user’s operating system username matches a valid Postgres username, that user can login with no further authentication.
+
+During the Postgres installation, an operating system user named postgres was created to correspond to the postgres PostgreSQL administrative user. You need to use this user to perform administrative tasks. You can use sudo and pass in the username with the -u option.
+
+Log into an interactive Postgres session by typing:
+
+    ➜ ~ sudo -u postgres psql
+
+First, you will create a database for the Django project. Each project should have its own isolated database for security reasons. We will call the database myproject in this guide, but it’s always better to select something more descriptive:
+
+    postgres=# CREATE DATABASE toman_interview;
+
+
+
+Note: Remember to end all commands at an SQL prompt with a semicolon.
+
+Next, you will create a database user which you will use to connect to and interact with the database. Set the password to something strong and secure:
+
+    postgres=# CREATE USER tomanuser WITH PASSWORD 'tomanpassword';
+
+Afterwards, you will modify a few of the connection parameters for the user you just created. This will speed up database operations so that the correct values do not have to be queried and set each time a connection is established.
+
+    postgres=# ALTER ROLE tomanuser SET client_encoding TO 'utf8';
+    postgres=# ALTER ROLE tomanuser SET default_transaction_isolation TO 'read committed';
```

Comment:
`read committed` :
a SELECT query (without a FOR UPDATE/SHARE clause) sees only data committed before the query began; it never sees either uncommitted data or changes committed during query execution by concurrent transactions. In effect, a SELECT query sees a snapshot of the database as of the instant the query begins to run.SELECT does see the effects of previous updates executed within its own transaction, even though they are not yet committed

`repeatable read`:
This level is different from Read Committed in that a query in a repeatable read transaction sees a snapshot as of the start of the first non-transaction-control statement in the transaction, not as of the start of the current statement within the transaction. Thus, successive SELECT commands within a single transaction see the same data, i.e., they do not see changes made by other transactions that committed after their own transaction started.

`serializable`
This level emulates serial transaction execution for all committed transactions; as if transactions had been executed one after another, serially, rather than concurrently. However, like the Repeatable Read level, applications using this level must be prepared to retry transactions due to serialization failures. In fact, this isolation level works exactly the same as Repeatable Read except that it monitors for conditions which could make execution of a concurrent set of serializable transactions behave in a manner inconsistent with all possible serial (one at a time) executions of those transactions. This monitoring does not introduce any blocking beyond that present in repeatable read, but there is some overhead to the monitoring, and detection of the conditions which could cause a serialization anomaly will trigger a serialization failure.